crates/sargon-uniffi/src/system/interactors/host_interactor.rs

```diff
@@ -0,0 +1,89 @@
+use crate::prelude::*;
+use sargon::KeyDerivationInteractor as InternalKeyDerivationInteractor;
+use sargon::KeyDerivationRequest as InternalKeyDerivationRequest;
+use sargon::KeyDerivationResponse as InternalKeyDerivationResponse;
+use sargon::Result as InternalResult;
+use sargon::SignInteractor as InternalSignInteractor;
+use sargon::Subintent as InternalSubintent;
+use sargon::SubintentHash as InternalSubintentHash;
+use sargon::TransactionIntent as InternalTransactionIntent;
+use sargon::TransactionIntentHash as InternalTransactionIntentHash;
+use sargon::UseFactorSourcesInteractor as InternalUseFactorSourcesInteractor;
+
+type InternalSignRequestForTransactionIntent =
+    sargon::SignRequest<InternalTransactionIntent>;
+type InternalSignWithFactorsOutcomeForTransactionIntent =
+    sargon::SignWithFactorsOutcome<InternalTransactionIntentHash>;
+type InternalSignRequestForSubintent = sargon::SignRequest<InternalSubintent>;
+type InternalSignWithFactorsOutcomeForSubintent =
+    sargon::SignWithFactorsOutcome<InternalSubintentHash>;
+
+/// Sargon os
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait HostInteractor: Send + Sync + std::fmt::Debug {
+    async fn sign_transactions(
+        &self,
+        request: SignRequestOfTransactionIntent,
+    ) -> SignWithFactorsOutcomeOfTransactionIntentHash;
+
+    async fn sign_subintents(
+        &self,
+        request: SignRequestOfSubintent,
+    ) -> SignWithFactorsOutcomeOfSubintentHash;
+
+    async fn derive_keys(
+        &self,
+        request: KeyDerivationRequest,
+    ) -> Result<KeyDerivationResponse>;
+}
+
+#[derive(Debug)]
+pub struct UseFactorSourcesInteractorAdapter {
+    pub wrapped: Arc<dyn HostInteractor>,
```

Comment:
rename `wrapped` -> `interactor` ? 

And maybe drop `Interactor` in `InteractorAdapter` ? I think `"Interactor Adatper" is a tautology.

Reply:
This is the same naming we use in adapters.

Reply:
See:
* `NetworkingDriverAdapter`
* `SecureStorageDriverAdapter`
* `EntropyProviderDriverAdapter`
* `HostInfoDriverAdapter`
* `LoggingDriverAdapter`
* `EventBusDriverAdapter`
* `FileSystemDriverAdapter`
* `UnsafeStorageDriverAdapter`
* `ProfileStateChangeDriverAdapter`

Reply:
Ah they were renamed when I was on vacation or something :P

Reply:
not renamed, always was wrapped.

Reply:
"always" you mean since UniFFI seperation, for some weeks?

Right so hmm Adaptors are quite invisible. They are not visible by Hosts and not used in crate `sargon` they merly "glue" the `trait `Foo` in `sargon` and `#[uniffi::with_forign]trait Foo` in `sargon-uniffi` together?

Reply:
was part of UniFFI PR. And yes that's their purpose, as the name Adaptor implies.